components/clarinet-format/src/formatter/mod.rs

```diff
@@ -0,0 +1,1220 @@
+        let result = format_with_default(src);
+        assert_eq!(src, result)
+    }
+    #[test]
+    fn closing_if_parens() {
+        let src = "(something (if (true) (list) (list 1 2 3)))";
+        let result = format_with_default(src);
+        let expected = r#"(something (if (true)
+    (list)
+    (list 1 2 3)
+  )
+)"#;
+        assert_eq!(expected, result);
+    }
+
+    #[test]
+    fn ok_map() {
+        let src = "(ok { a: b, c: d })";
+        let result = format_with_default(src);
+        let expected = r#"(ok {
+  a: b,
+  c: d
+})"#;
+        assert_eq!(expected, result);
+    }
+
+    #[test]
+    fn if_let_if() {
+        let src = r#"(if (true)
+  (let (
+    (a (if (true)
+        (list)
+        (list)
+    ))
+  )
+    (list)
+  )
+  (list)
+)"#;
+        let result = format_with_default(src);
+        assert_eq!(src, result);
+    }
+
+    #[test]
+    fn map_in_map() {
+        let src = "(ok { a: b, ctx: { a: b, c: d }})";
+        let result = format_with_default(src);
+        let expected = r#"(ok {
+    a: b,
+    ctx: {
+      a: b,
+      c: d,
+    },
+  }
+)"#;
+        assert_eq!(result, expected);
+    }
+
+    #[test]
+    fn old_tuple() {
+        let src = r#"(tuple
+  (a uint)
+  (b uint) ;; comment
+  (c bool)
+)"#;
+        let result = format_with_default(src);
+        let expected = r#"{
+  a: uint,
+  b: uint, ;; comment
+  c: bool
+}"#;
+        assert_eq!(result, expected);
+    }
+
+    #[test]
+    fn unwrap_wrapped_lines() {
+        let src = r#"(new-available-ids
+  (if (is-eq no-to-treasury u0)
+    (var-get available-ids)
+    (unwrap-panic
+      (as-max-len? (concat (var-get available-ids) ids-to-treasury) u10000)
+    )
+  )
+)"#;
+        let result = format_with_default(src);
+        assert_eq!(src, result);
+    }
+
+    #[test]
+    fn test_irl_contracts() {
+        let golden_dir = "./tests/golden";
+        let intended_dir = "./tests/golden-intended";
+
+        // Iterate over files in the golden directory
+        for entry in fs::read_dir(golden_dir).expect("Failed to read golden directory") {
+            let entry = entry.expect("Failed to read directory entry");
+            let path = entry.path();
+
+            if path.is_file() {
+                let src = fs::read_to_string(&path).expect("Failed to read source file");
+
+                let file_name = path.file_name().expect("Failed to get file name");
+                let intended_path = Path::new(intended_dir).join(file_name);
+
+                let intended =
+                    fs::read_to_string(&intended_path).expect("Failed to read intended file");
+
+                // Apply formatting and compare
+                let result = format_file_with_metadata(&src);
+                pretty_assertions::assert_eq!(
+                    result,
+                    intended,
+                    "Mismatch in file: {:?}",
+                    file_name
+                );
+            }
```

Comment:
Ideally, integration tests should leave in the [`./tests` directory](https://doc.rust-lang.org/book/ch11-03-test-organization.html#the-tests-directory) (close the golden/golden-intended contracts actually)
We don't do a very good job at doin this in the repo, but we aim to do better.

(Only if this is not too involved to move this test)